src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/PublicVisitsTest.kt

```diff
@@ -0,0 +1,149 @@
+  private lateinit var visitAssertHelper: VisitAssertHelper
+
+  @SpyBean
+  private lateinit var telemetryClient: TelemetryClient
+
+  private lateinit var otherSessionTemplate: SessionTemplate
+
+  private lateinit var visit: Visit
+  private lateinit var visitInDifferentPrison: Visit
+  private lateinit var visitWithOtherBooker: Visit
+
+  @BeforeEach
+  internal fun createVisits() {
+    otherSessionTemplate = sessionTemplateEntityHelper.create(prisonCode = "AWE")
+
+    visitInDifferentPrison = createApplicationAndVisit(prisonerId = "AA0000A", slotDate = startDate.plusWeeks(1), sessionTemplate = otherSessionTemplate, userType = PUBLIC)
+    visitInDifferentPrison = visitEntityHelper.save(visitInDifferentPrison)
+
+    eventAuditEntityHelper.createForVisitAndApplication(visitInDifferentPrison, actionedByValue = "aTestRef", type = listOf(RESERVED_VISIT, BOOKED_VISIT))
+
+    visit = createApplicationAndVisit(prisonerId = "FF0000CC", slotDate = startDate.plusDays(2), sessionTemplate = sessionTemplateDefault, userType = PUBLIC)
+    visit = visitEntityHelper.save(visit)
+
+    eventAuditEntityHelper.createForVisitAndApplication(visit, actionedByValue = "aTestRef", type = listOf(RESERVED_VISIT, BOOKED_VISIT))
+
+    var visitInPast = createApplicationAndVisit(prisonerId = "FF0000CC", slotDate = LocalDate.now().minusDays(1), sessionTemplate = sessionTemplateDefault, userType = PUBLIC)
+    visitInPast = visitEntityHelper.save(visitInPast)
+
+    eventAuditEntityHelper.createForVisitAndApplication(visitInPast, actionedByValue = "aTestRef", type = listOf(RESERVED_VISIT, BOOKED_VISIT))
+
+    var visitBookerByStaff = createApplicationAndVisit(prisonerId = "FF0000CC", slotDate = LocalDate.now().minusDays(1), sessionTemplate = sessionTemplateDefault, userType = STAFF)
+    visitBookerByStaff = visitEntityHelper.save(visitBookerByStaff)
+
+    eventAuditEntityHelper.createForVisitAndApplication(visitBookerByStaff, actionedByValue = "aTestRef", type = listOf(RESERVED_VISIT, BOOKED_VISIT))
+
+    var visitCancelled = createApplicationAndVisit(prisonerId = "FF0000CC", slotDate = startDate.plusDays(3), sessionTemplate = sessionTemplateDefault, visitStatus = CANCELLED, userType = PUBLIC)
+    visitCancelled.outcomeStatus = OutcomeStatus.CANCELLATION
+    visitCancelled = visitEntityHelper.save(visitCancelled)
+
+    eventAuditEntityHelper.createForVisitAndApplication(visitCancelled, actionedByValue = "aTestRef", type = listOf(RESERVED_VISIT, BOOKED_VISIT, CANCELLED_VISIT))
+
+    visitWithOtherBooker = createApplicationAndVisit(prisonerId = "AA0000A", slotDate = startDate.plusWeeks(1), sessionTemplate = otherSessionTemplate, userType = PUBLIC)
+    visitWithOtherBooker = visitEntityHelper.save(visitWithOtherBooker)
+
+    eventAuditEntityHelper.createForVisitAndApplication(visitWithOtherBooker, actionedByValue = "aOtherTestRef", type = listOf(RESERVED_VISIT, BOOKED_VISIT))
+  }
+
+  @Test
+  fun `get public booked visit by booker reference aTestRef`() {
```

Comment:
Could this describe what the test is doing? Instead of just getting by reference aTestRef?